src/rules/firestore-transaction-reads-before-writes.ts

```
@@ -0,0 +1,300 @@
+        VariableDeclarator(node) {
+          if (
+            node.id.type === AST_NODE_TYPES.Identifier &&
+            node.init &&
+            node.init.type === AST_NODE_TYPES.Identifier
+          ) {
+            const transactionScope = findTransactionScope(node);
+            if (transactionScope) {
+              const scopeInfo = transactionScopes.get(transactionScope);
+              if (
+                scopeInfo &&
+                scopeInfo.transactionVariables.has(node.init.name)
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)